### PR TITLE
Refine proto buffer describing PPA results.

### PR DESCRIPTION
**Pull request by hzeller**
This standardizes units for power (watts), time (picoseconds), and length units (um or um2) and breaks out more information that might be available from the EDA tools.

Compared to the original proto, there are so many changes, that this is a new protocol buffer, based on the old one. So to not clash with the original, this uses a new name PowerPerformanceAreaProto; the old proto needed a small change in the name of a type (because we really want to use the name 'Power' in the new one with a different definition), but it is fully binary compatible to the one before.

TL;DR:
  * old: hdl.ppa.PerformancePowerAreaProto
  * new: hdl.ppa.PowerPerformanceAreaProto

Next step will be to adapt the bazel rules currently generating the old proto to create the new one.

Once this new one is established, the old one can be deprecated and removed quickly after last use.

### CODE REVIEW

File: synthesis/power_performance_area.proto [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+  optional int32 num_combinational = 11;
+
+  // The number of non-timing buffers.
+  optional int32 num_buffers = 12;
+
+  // The number of timing buffers.
+  optional int32 num_timing_buffers = 13;
+
+  // The number of flops.
+  optional int32 num_flops = 14;
```
Comment (mikesinouye) [inline synthesis/power_performance_area.proto:76]: I am reversing what I said earlier, perhaps "sequential" would be better instead of "flops". There are some sequential cells like synchronizers that are not flops, which may cause the total to add to total_cells. Sequential seems more generic, so we could be less precise so things still add up to the total.

File: synthesis/power_performance_area.proto [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+  map<string, float> celltype_fractions = 20;
+}
+
+message Performance {
+  // Desired clock frequency this PPA metric was generated for.
+  // It influences the choice of area-impacting optimization, so this
+  // helps to understand area size in that context.
+  optional int32 clock_period_ps = 1;
+
+  // Longest path in picoseconds between any two flop-boundaries.
```
Comment (mikesinouye) [inline synthesis/power_performance_area.proto:91]: I would omit the "two flop boundaries" part, as I think we would still want to report an in-to-FF and a FF-to-out delay as the critical path, if it is.

File: synthesis/power_performance_area.proto [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+  optional int32 num_hold_violations = 11;
+  optional int32 num_slew_violations = 12;
+  optional int32 num_fanout_violations = 13;
+  optional int32 num_capacitance_violations = 14;
+
+  // The number of cells in the longest topological path.
+  optional int32 longest_topological_path = 15;
+
+  // Number of cells in the critical path
+  optional int32 critical_path_cells = 16;
```
Comment (mikesinouye) [inline synthesis/power_performance_area.proto:136]: How is this different than above? Does longest path mean in-to-out? If so, I would specify that int he comment, as a I was assuming "longest_topological_path" was a constrained path.

**Review by mikesinouye — CHANGES_REQUESTED** [review]

File: synthesis/power_performance_area.proto (hunk 1 above)
Comment (hzeller) [reply synthesis/power_performance_area.proto]: done.

File: synthesis/power_performance_area.proto (hunk 2 above)
Comment (hzeller) [reply synthesis/power_performance_area.proto]: Done.

File: synthesis/power_performance_area.proto (hunk 3 above)
Comment (hzeller) [reply synthesis/power_performance_area.proto:137]: Clarified in comment that the criticial path one is based on timing.

**Review by mikesinouye — APPROVED** [review]
